controllers/MobileAppController.php

```diff
@@ -2,45 +2,23 @@
 
 namespace humhub\modules\fcmPush\controllers;
 
-use humhub\modules\admin\components\Controller;
-use humhub\modules\admin\notifications\NewVersionAvailable;
-use humhub\modules\fcmPush\models\FcmUser;
-use humhub\modules\user\models\User;
+use humhub\components\Controller;
+use humhub\libs\Html;
+use humhub\modules\fcmPush\helpers\MobileAppHelper;
 use Yii;
 
 class MobileAppController extends Controller
 {
-    /**
-     * Renders the index view for the module
-     *
-     * @return string
-     */
-    public function actionIndex()
+    public function actionInstanceOpener()
     {
-
-        if (Yii::$app->request->get('triggerNotification') == 1) {
-
-            /** @var User $user */
-            $user = Yii::$app->user->getIdentity();
-
-            $updateNotification = new NewVersionAvailable();
-            $updateNotification->sendBulk(User::find()->where(['user.id' => $user->id]));
-            $this->view->setStatusMessage('success', 'Notification queued!');
-            return $this->redirect('index');
-        }
-
-        if (Yii::$app->request->get('deleteToken') != "") {
-            $t = FcmUser::findOne(['id' => Yii::$app->request->get('deleteToken')]);
-            if ($t->delete() !== false) {
-                $this->view->setStatusMessage('success', 'Token deleted!');
-                return $this->redirect('index');
-            } else {
-                $this->view->setStatusMessage('warning', 'Token NOT deleted!');
-                return $this->redirect('index');
-            }
-        }
-
-        return $this->render('index');
+        // Send to the mobile app to display the instance opener
+        Yii::$app->session->set(MobileAppHelper::SESSION_VAR_SHOW_OPENER, 1);
+
+        // Stay on the same page, because when we come back from the mobile app to this instance
+        // Force full page refresh to trigger the onLayoutAddonInit event
+        return $this->renderContent(sprintf(
+            '<script ' . Html::nonce() . '>window.location.href = "%s";</script>',
```

Comment:
@marc-farre Would this a endless loop when the mobile app not working correctly?

Reply:
No, because we just post a JS a hideOpener message to the flutterChannel and go back to the previous page (the referrer) using:
```
return $this->htmlRedirect(Yii::$app->request->referrer);
```